### DISCUSSION TITLE
Roadmap for built-in connectors for Logic Apps Standard

### DISCUSSION THREAD

**Discussion by divyaswarnkar** [Azure/logicapps#530] [Upcoming connectors]
# Roadmap for Built-in connectors for Logic Apps Standard

Logic Apps Standard supports all [managed connectors](https://docs.microsoft.com/en-us/connectors/connector-reference/connector-reference-logicapps-connectors) available from Logic Apps consumption. This is the list of [built-in connectors](https://docs.microsoft.com/en-us/azure/connectors/compare-built-in-azure-connectors) available and planned for the calendar year. If you need something that’s not in the list, please [share your ideas](https://github.com/Azure/logicapps/discussions/categories/ideas)!

\*Please note that these timelines are soft targets, and the actual dates may shift. However, we will make best attempt to keep this list and timelines up to date.

\* Full functional parity implies parity with existing managed connectors

\* 9/13/2022 - Updated the list with Service Providers released until end of August

\* 11/22/2022 - Updated the list with Service Providers released until early Nov. The most recent updates include support for SQL trigger, File System connector (only on ASE V3), XML Transform with XSLT 2.0/3.0 and custom assemblies, Integration Account support for AS2 and other actions, Large message support for SFTP, FTP and File system

\* 1/23/2023 – Added the list of built-in connectors that recently moved from Preview to GA. Also updated the status of SAP, X12

\* 4/26/2023 – Added the list of built-in connectors that recently moved from Preview to GA. At this point, we are at full feature parity between ISE core connectors and LA Standard built-in connectors

Here is the list of built-in connectors that are in now Generally Available (GA)

-   EventHub
-   Azure Automation
-   Key Vault
-   SQL Server
-   SFTP
-   FTP
-   SMTP
-   Azure Blob Storage
-   Azure Queue Storage
-   Azure File Storage
-   Azure Table Storage
-   Event Hub
-   Service Bus (more operations in pipeline)
-   XML Operations
-   Liquid Operations
-   Event Grid Publish
-   Flat File
-   MQ
-   AS2
-   APIM
-  Batch Operations
-  RosettaNet
-  Integration Account lookup
- SAP

These connectors are at full parity and in limited GA (GA level support available to customers on request)
-  Azure Cosmos

Here is the roadmap for the upcoming built-in connectors.

| Built-in Connector            | What’s coming next                       | Timelines | Note                                                                    |
|-------------------------------|------------------------------------------|-----------|-------------------------------------------------------------------------|
| DocumentDB/Cosmos (Preview)   | GA                                       | CY24  H2  |               |
| X12 (Preview)            | GA                                  | CY24 Q1   |                           |
| EDIFACT (Preview)        | GA                                  | CY24 Q1     |                       |
| Event Grid (Not Released)     | Planned                                  | TBA       |                                                                         |

\*\* GA - Generally Available  
\*\* TBA - To be Announced  
\*\* VNET - Virtual Network

**Comment by ajseppala** [comment]
File System - does this have functional parity with BizTalk file adapter, i.e. can be pointed to a file share without OPDGW?

**Reply by divyaswarnkar** [reply]
Hi @ajseppala , the built-in connector will have functional parity with Logic Apps managed connector. Like all other built-in connectors, File System will support direct VNET access and you would be able to access file shares without the need for on-prem data gateway. Let us know what operations are more important for you.

**Reply by ajseppala** [reply]
Alright, sounds good. Just to double check, Arc-enabled LA on on-premises k8s will support file system access?

**Reply by raunaknarooka** [reply]
when will this File system connector be released ? :)

**Reply by ErikMogensen** [reply]
> Let us know what operations are more important for you.

@divyaswarnkar, it was some ago since you asked so I realize it may not matter anymore. Anyway, for us the following operation are more important:
1. Create file
2. List files
3. Get file content
4. Delete file

**Reply by gopalPrasath** [reply]
> Hi @ajseppala , the built-in connector will have functional parity with Logic Apps managed connector. Like all other built-in connectors, File System will support direct VNET access and you would be able to access file shares without the need for on-prem data gateway. Let us know what operations are more important for you.

I am trying to connect to an on prem Network File Share from my logic app hosted in an internal network (VNET).
I am using a standard single tenant logic app.

I tried to establish a connection using the managed File System connector via OPDG and I was able to connect to the File Share.

Since I am connecting to an on prem File Share I want to connect through the in-app connector as in-app connectors are faster compared to managed connectors.
But on trying to establish a connection using an in-app connector I'm getting an exception saying **"The service provider action failed with error code 'ServiceOperationFailed' and error message 'The username or password is incorrect."**

FYI, I'm using the same account and credentials which I used to connect through managed connector to establish the connection from the in-app connector.

Please let me know why this issue is happening, is there a known issue with this connector or should I be following a different approach to get this done.

**Reply by anandgmenon** [reply]
@gopalPrasath please check out this blog post https://techcommunity.microsoft.com/t5/azure-integration-services-blog/using-file-system-connector-in-logic-apps-standard/ba-p/3711426. We have mentioned the common issues and troubleshooting when using the connector.

**Reply by gopalPrasath** [reply]
The above link is not available anymore it says "The requested content does not exist". Can you please check?

**Reply by OmarAbuArisheh** [reply]
> The above link is not available anymore it says "The requested content does not exist". Can you please check?
@gopalPrasath 

https://techcommunity.microsoft.com/blog/integrationsonazureblog/using-file-system-connector-in-logic-apps-standard/3711426

**Comment by your-azure-coach** [comment]
Hey, are there any plan to build an Oracle connector?

**Reply by divyaswarnkar** [reply]
Hi @your-azure-coach , we do have plans to build Oracle connector as a built-in, however I do not have any timelines to share yet. We are planning to release JDBC connector as built-in in CY22 Q3, that should also allow you to connect to Oracle DB without the need for data gateway.

**Reply by FlingersDH7** [reply]
Do you have an example on how such a conenction looks like?
Which driver do we need to use for example?

**Reply by nikolaiblackie** [reply]
@divyaswarnkar any plans to fix missing out parameter issues with data gateway based connector? Its a pretty big limitation when trying to migrate solutions to Azure.

**Reply by danhas14** [reply]
> Do you have an example on how such a conenction looks like? Which driver do we need to use for example?

@divyaswarnkar  I'm also trying to use the JDBC connector to connect to a Postgres database on my internal VNet. I've set the Java version in the configuration to Java 17, but I'm getting the error below when I try to run my workflow. Are there any docs I'm missing on where to set the driver?

{
  "code": "ServiceProviderActionFailed",
  "message": "The service provider action failed with error code 'ServiceOperationFailed' and error message 'No suitable driver found..
}

**Reply by danhas14** [reply]
I just saw this new blog on this topic @FlingersDH7 @divyaswarnkar  https://techcommunity.microsoft.com/t5/integrations-on-azure-blog/connecting-to-a-relational-database-using-jdbc-in-azure-logic/ba-p/3681950

**Reply by danhas14** [reply]
> Hi, we do have plans to build Oracle connector as a built-in, however I do not have any timelines to share yet. We are planning to release JDBC connector as built-in in CY22 Q3, that should also allow you to connect to Oracle DB without the need for data gateway.

Is there a target GA date for the JDBC adapter?

**Comment by dgrospelier** [comment]
Hi,
I want to have a confirmation, because I'm not able to find it on the documentation: when we use managed connectors in Logic App Standard hosted in a ASEv3, are those managed connectors hosted outside of the ASEv3, meaning that they are executed in the multi-tenant architecture and are using the same outbound IPs as the managed connectors in Logic App Consumptions?

**Reply by divyaswarnkar** [reply]
Hi @dgrospelier , managed connectors (aka Azure connectors) are hosted and executed in the multi-tenant environment regardless of the SKU (consumption or Standard or ASE v3). So your understanding is correct - you'd use the same outbound IPs or service tags to whitelist the calls in ASE v3. Hope that helps

**Comment by dgrospelier** [comment]
Hi,
I have a question regarding the Service Bus built-in connector.
Today, in StateLess workflows, we have the "When a message are available in a queue (or topic subscription)" with the "peek and lock" behaviour (which does not exists in StateFull at the time writing this message).
I want to know if by using this trigger, we have the "Durable Terminus" feature included, I mean: we no longer need to renew lock tokens (with Service Bus Premium SKU)?

**Comment by huntr01** [comment]
Hi,
Is there any update on the timelines for those connectors marked as CY22 Q2? 

Thanks

**Reply by divyaswarnkar** [reply]
They are all code complete and getting deployed. I am going to update with exact timelines soon

**Reply by dgrospelier** [reply]
@divyaswarnkar: what does it mean "not release" in front of the name of several connectors?

**Reply by divyaswarnkar** [reply]
Not released means that we have not released the connector. There are connectors which are released but do not have full functional parity with consumption/ISE. This table lists both the types of connectors and what's coming next is where we try to give you an idea of what capability is coming for that connector. hope this helps.

**Reply by ErikMogensen** [reply]
@divyaswarnkar, is the status after "Not released" preview?

**Comment by msitsoftware** [comment]
Hi, is there a plan to add a Connector to support OFTP2? [RFC5024](https://www.rfc-editor.org/rfc/rfc5024.html)

**Reply by divyaswarnkar** [reply]
@msitsoftware not at the moment. How important is it for you ?

**Reply by msitsoftware** [reply]
It is a key feature to decide to go with AIS or with another Integration Stack.

**Reply by maxlisi** [reply]
@msitsoftware no news after a year? :)

**Comment by mhireholi** [comment]
@divyaswarnkar can you please update the list?

**Reply by ziv0309** [reply]
@divyaswarnkar - I am also interested in an updated list. Specifically, for "Azure Queue" which shows "August end" (in the past)

**Reply by divyaswarnkar** [reply]
@ziv0309 please see updated list. Azure Queue is in Preview now.

**Comment by dgrospelier** [comment]
About the Service Bus connector, I have two questions please:

- Are you going to release the connector with StateFull Logic Apps?
- Is this connector able to handle "unlimited" duration of lock tokens in PeekLock?

**Reply by divyaswarnkar** [reply]
@dgrospelier Service Bus connector is available with LA Standard Stateful workflows as well. The peek lock actions are only available with Stateless workflows though. Do you need to use SB with stateful only ?

**Reply by divyaswarnkar** [reply]
Adding @apranaseth to answer about unlimited duration

**Reply by dgrospelier** [reply]
@divyaswarnkar: yes we do need the peek&lock in StateFull, today we need to use the Managed Connector and thus we need to connect to our namespace via the public endpoint, and this is something that is not compliant to our customer security rules.

**Reply by pacodelacruz** [reply]
Hi @divyaswarnkar , thanks for maintaining this roadmap. 
Regarding the Service Bus trigger with peek-lock. Peek lock is very important to maintain transactions in our integration solutions. Additionally, one of the key differentiators of built-in connectors is the networking capabilities so we can connect to SB via a Private Endpoint. Furthermore, another key value that Logic Apps provides compared to Azure Functions is the the out-of-the-box monitoring capabilities. With stateless workflows however, we lose most of that. Are there any plans to support peek-lock in the SB trigger for stateful workflows in the near future? It'd be great if that is stated in this roadmap. 
Thanks,

**Comment by Msal81** [comment]
@divyaswarnkar  ****Any update on Batch and X12 for standard?  Really need this!!****  Trying to batch encoded x12 messages into blobs without having to create dozens of consumption logic apps (two per trading partner)

**Reply by Msal81** [reply]
> @divyaswarnkar ****Any update on Batch and X12 for standard? Really need this!!**** Trying to batch encoded x12 messages into blobs without having to create dozens of consumption logic apps (two per trading partner)

Need to do the following in standard workflow (instead of consumption LAPP as illustrated below):
![image](https://user-images.githubusercontent.com/113480770/190026184-8879ddf3-0bff-4655-8560-58ae32ba4b84.png)

**Reply by divyaswarnkar** [reply]
@Msal81 we have completed backend work for batch action and UI work is pending. I expect this action to release early CY22 Q4. Regarding X12 connector, you should be able to use the managed connector. Please let me know if that doesnt work for you.

**Reply by Msal81** [reply]
Hi @divyaswarnkar  - Any update on this?  We are attempting to invoke a workflow from consumption based logic app that we have to use for this one action - but are unable to do so as our standard logic app is configured for private endpoint only

**Comment by FlingersDH7** [comment]
We are using the build in Service Bus connector because we use Logic Apps Standard on an ASE with Vnets. Managed Connectors is a pain to get running in this config. We got the SFTP/SSH and SharePoint Connector running, but the SB Managed Connector does not want to work. This is why we choose (for now)  to use the build in connector.

But we are running in some issues regarding the following.

We want to process related messages in  a specific order, by using a sequential convoy. This is a know working pattern in Logic Apps Consumption, but how do we use this pattern with the build in connector.
We know that  the preview version of the stateless SB connector has peek/lock which is needed to create a convoy, but this trigger only works when you disable  Split-On. And by disabling Split-On, we get an array of messages. Which is something we do not want. This is our first problem.

After that issue, we are at a complete dead end on how even to implement an convoy.
On major part of a convoy is that  after you receive a message  on a queue/topic, you begin to read messages from a queue/ topic  in a loop to determine if there are more messages of the same session to process. This way you can run the logic app until all messages of the same session are processed in the correct order.
In the build in connector, there is no read from queue/topic option available.

Is this pattern even possible with the current state of the build in Connector? Or are there other solutions for processing related messages in  a specific order using the build in connector?
And is there a solution with the split on issue?

**Reply by John-Bosch** [reply]
I would very much like this question answered. At the moment, I can not find a way to get the sequential convoy pattern to work using GA operations.

I would also like to know who locking is handled and whether locks need to be renewed within the workflow, or if they are handled under the covers by the connector code itself. There does not appear to be a "renew session lock" action at all, in fact!

**Reply by divyaswarnkar** [reply]
We have recently released a couple of enahancements to Service Bus. These should address your questions. Please check out the blog post

https://aka.ms/sbconcurrency
https://aka.ms/sbsessions

**Reply by bally-singh** [reply]
Hi @divyaswarnkar 

Any idea when this will be GA'd?

https://techcommunity.microsoft.com/t5/azure-integration-services-blog/session-support-for-service-bus-built-in-connector-logic-apps/ba-p/4034074

**Comment by FlingersDH7** [comment]
Another Question.
Is support for ActiveMQ also considered?

**Reply by mohit-jagtiani** [reply]
Any response?

**Comment by crsnt** [comment]
SFTP (Preview) | Large message support

What file size limit does SFTP support now and what would it be after release? Is this going to GA soon too? 
We are looking to use this after we establish ExpressRoute to on-prem.

Thanks!

**Reply by S-Jindal** [reply]
We have released large content support in SFTP actions 2 months ago. See this blog post for complete details - https://techcommunity.microsoft.com/t5/integrations-on-azure-blog/feature-available-large-content-transfer-using-built-in/ba-p/3668649

Sftp connector is released for GA

**Comment by KritikaSingh89** [comment]
Hi @divyaswarnkar, 

Can you please update when we can expect SAP and SFTP-SSH In-Built Connector available for Logic Apps Standard?

Thanks & Regards,
Kritika Singh

**Reply by peeyushverma256** [reply]
Hi @divyaswarnkar,
Any update on the same(SAP & SFTP-SSH) Built-In connectors for Logic Apps Standard?

Thanks & Regards,
Peeyush

**Reply by dgrospelier** [reply]
Hi @divyaswarnkar,
The SAP built in connector appears to be "not released" in this article, but we can see it in the list of Logic Apps connectors in our tenant, is it then released?

**Reply by daviburg** [reply]
The SAP built-in connector is now not only released but furthermore in General Availability.

**Comment by raunaknarooka** [comment]
I see a filesystem connector in built in connectors 
![image](https://user-images.githubusercontent.com/24993027/199395843-b275d703-dc9d-43aa-bab6-7565791cb815.png)

When i try using it,  i get the error as:
![image](https://user-images.githubusercontent.com/24993027/199395990-312934e0-681d-4422-8337-80cbd2cb9fe2.png)

In the timelines, I see that it will be released in Q4 2022. Is it working now?
This connector will really be useful for us to replace Biztalk in many scenarios.

**Reply by KritikaSingh89** [reply]
Hi,

Are you using Logic App Standard ? If yes, which environment are using ASE or normal App Service Plan ?

**Reply by raunaknarooka** [reply]
normal ASP and yes i am using LA standard

**Reply by KritikaSingh89** [reply]
Also, there is problem when we use FILE SYSTEM. The user name and password used to connect fileshare is automatically added to the environment variable and cannot be changed using parameter or appsettings file for different environment.

**Comment by shyamazure** [comment]
@divyaswarnkar : As per table it states Q42022 for below connectors , can we have update on when these will be in GA as we have production systems/services  lined up for utilizing these connectors

SFTP
Eventhub
ServiceBus
Blob
Table
File
Queue
SQL

**Comment by JackXueIndiana** [comment]
@divyaswarnkar Why I did not see LAS built-in connector for Azure Functions? Per this document https://learn.microsoft.com/en-us/azure/connectors/built-in, it should be already available in the Service Provider model. Please clarify. Thanks.

**Comment by manjunath-hireholi** [comment]
@divyaswarnkar : Is the timeline up-to-date on the connectors? will they all be GA by the end of this month?

**Reply by mhireholi** [reply]
@divyaswarnkar any update?

**Reply by UmairSyed** [reply]
any updates? we need to connect to table storage from logic apps. What are the alternatives if we want to use managed identity of logic app to access table storage?

**Comment by KiwiGitCat** [comment]
@divyaswarnkar Do we have any updates on the connectors due for GA in CY22 Q4?
Thanks

**Reply by divyaswarnkar** [reply]
@KiwiGitCat updated the list

**Comment by mohit-jagtiani** [comment]
@divyaswarnkar where can I find documentation on ServiceBus connector (operations available and in pipeline)? Also How do I request new connector? I wish to send and receive message from activemq client application. Do I still need to host a service bus in my tenant or can it be done via an in-app connector (if in development pipeline)?

**Reply by mohit-jagtiani** [reply]
@divyaswarnkar any updates?

**Comment by mohit-jagtiani** [comment]
Also when will the preview designer be made generally available?

**Comment by behlerdavid** [comment]
the SAP connector is not supporting Personal Security Environment (PSE) until now. This is a must have for Secure Network Communications (SNC). Without that no switch from logic app consumption is possible.

**Reply by behlerdavid** [reply]
Just saw that the "Authentication Type" of in-built SAP connector has a drop down menu, to select SNC. That should be fine for us then.

**Reply by daviburg** [reply]
Indeed, the built-in SAP connector had Personal Security Environment (PSE) support for Secure Network Communications (SNC) added. This feature is GA as well.

**Comment by KritikaSingh89** [comment]
Hi @divyaswarnkar,

Can you please update if SAP Built-In Connector is now available GA.

Thanks & Regards,
Kritika Singh

**Reply by daviburg** [reply]
The SAP Built-In connector is now GA indeed.

**Comment by Chrissie** [comment]
Hi @divyaswarnkar, Do you have any update on the Event Grid (Subscribe) connector availability?

**Comment by Dhana1207** [comment]
@divyaswarnkar  In the Logic app standard workflow we are using built-in service bus connector with peek lock trigger. Read message and send message to the queue actions are working fine but
when we are completing the message in the queue in the logic app workflow we are getting below error.

Error :

  "code": "ServiceProviderActionFailed",
  "message": **"The service provider action failed with error code 'VNetPrivatePortsNotConfigured' and error message 'Using this service provider action in stateful workflows requires VNet integration and the site configuration vnetPrivatePortsCount to be set to a value greater than or equal to 2.'."**

Can you please someone help us with the issue.

**Reply by jaapvandijke** [reply]
_"Service Bus and SAP built-in connectors in Logic App Standard contain many triggers and actions that were previously restricted to stateless Logic Apps only. Beginning in April, these triggers and actions are now available for use in stateful Logic Apps. This article explains the requirements needed to successfully use this new feature."_

Reference: https://techcommunity.microsoft.com/t5/azure-integration-services-blog/enabling-service-bus-and-sap-built-in-connectors-for-stateful/ba-p/3820381

**Reply by AbhijatSaxena** [reply]
To summarize - please enable "VNet Integration" in your logic app

Refer - https://techcommunity.microsoft.com/t5/azure-integration-services-blog/enabling-service-bus-and-sap-built-in-connectors-for-stateful/ba-p/3820381

We were facing similar issue with Complete Message & Dead letter message actions, it's resolved now.

**Reply by swethamaram984** [reply]
yes, after enabling vnet integration out issue got resolved, but some times we used to get this error even after enabling the Vnet integration as well. Any idea why is it happening or any one else is facing this issue.

**Reply by Dhana1207** [reply]
@AbhijatSaxena @jaapvandijke @divyaswarnkar  

in Logicapp std , we were able to resolve the issue by enabling Vnet and setting the configuration as suggested. But We have a requirement to containerize same standard logic app workflows which uses service bus built in connector . we were able to push the image and test, but Complete message action is failing with bad request error. As a solution suggested for Standard logic apps , we tried implementing the same solution for Container instance by adding vnet and subnet, apart from this we need to set vnetPrivatePortCount to 2 but this property is not available for the container instance. Any suggestions would be helpful.

**Comment by swethamaram984** [comment]
@divyaswarnkar , for the built in actions is it possible to use expressions to set retrycount and retryinterval in Standard logic apps. we tried to assign parameter value but its not allowing. is there any way to use expressions or is it not allowed? kindly confirm.

**Comment by GaurSumit** [comment]
@divyaswarnkar  - Could you have the article updated with the connector that were in preview but now are in GA, and also if any new connectors being added to road map for next year?

**Comment by lilan123** [comment]
@divyaswarnkar, Is there any plan for a "When a message is received in a topic subscription (peek-lock)" trigger for the service bus built-in connector in the stateful workflows?

**Comment by TWolversonReply** [comment]
The [documentation](https://learn.microsoft.com/en-us/azure/logic-apps/logic-apps-enterprise-integration-flatfile?tabs=consumption) says Flat File for Standard is still in preview, whereas the initial post here says it's GA - which is it?

**Reply by TWolversonReply** [reply]
Documentation has now been updated.

**Comment by yashwantsahu05** [comment]
When File system build in connector will be GA?
It's will overcome most of BizTalk scenario, As it's in preview are are unable to take call to use it or not for our integration scenario.

**Reply by yashwantsahu05** [reply]
@divyaswarnkar Can we get update on this please?

**Comment by bankarswapnil** [comment]
Hello Team / @divyaswarnkar ,
After going through the SAP adapter and Logic Apps documentation, I could not confirm on following,
If I have a on-premises based SAP system then can I use Azure Standard logic apps to connect to SAP system using Managed SAP connector via Azure on-premises data gateway(+NCo3.0 libs installed here) ?
Whereas I have already confirmed Consumption Logic Apps can connect to on-prem SAP system via Azure on-premises DataGateway(+NCo3.0 libs installed here) using Managed SAP connector.

Being said that can we conclude,
To connect to on-prem SAP, Consumption logic Apps with Managed SAP connector is the only option ?

Let me know.

Thanx,
-Swapnil

**Reply by daviburg** [reply]
To connect to an on-premises based SAP system from a Logic Apps Standard either of the Managed SAP connector via OPDG (+NCo 3.1 libs) or built-in SAP connector, V-NET peering (+NCo 3.1 libs) can be used. Solutions such as Express Route can be used for peering on-premises network to Azure network.

**Comment by Squarepour** [comment]
Any chance the Blob storage connector and actions could be enhanced to permit the writing/creation of **metadata** on the blobs you're uploading/managing?  

I'm stuck without this.  HTTP PUT works to permit creating metadata, but SAS token management to the blob storage container is manual as far as I can find.  Permanent HTTP Authorization header including the container key feels risky.  

Thanks!

**Reply by daviburg** [reply]
I will give a heads up to the built-in Blob storage connector owner(s) regarding your feature suggestion.

**Reply by Squarepour** [reply]
Awesome!  Thank you.  Writing blob metadata with the convenience of Managed Identity authorization (from Logic App to Storage Account) would be a huge boost.

**Comment by KavyaMuniswamappa** [comment]
Do we have the built-in connectors for Azure Data Explorer in Logic App(Standard) ?

**Comment by rubenaster** [comment]
A client of mine has the requirement to create folders via the File System connector (Logic App Standard on ASE v3 using built-in connector). Currently only the following actions are supported:

- Create, get, append, update, and delete files.
- List files in folders or root folders.
- Get file content and metadata.

(https://learn.microsoft.com/en-us/azure/connectors/file-system?tabs=standard)

Do you have creating folders already on your roadmap for that connector?

**Comment by raunaknarooka** [comment]
Hello @divyaswarnkar,

Regarding the In app SFTP connector, there is a setting in host.json file called 'ServiceProviders.MaximumAllowedTriggerStateSizeInKB' which by default is set to around 1000 files. So this means at a time, the SFTP connector can only read 1000 files. If we have more than 1000 files, the SFTP connector does not pcik up a file and gives an error in the trigger history as ServiceProviderTriggerStateTooLarge
But if we have a service window for example; we get more than 1000 files, then the SFTP connector wont work unless we change the host .json file setting. May be the behaviour should be that it picks first 1000 files and then waits and picks the remaining files?

**Reply by anandgmenon** [reply]
The `ServiceProviders.MaximumAllowedTriggerStateSizeInKB` host setting is not a SFTP connector specific setting but rather used by bunch of polling triggers. For such triggers the logic apps runtime maintains a trigger state internally (this is not exposed to the users) and we have a default limit of 10KB and its not just based on the number of files. In case of SFTP trigger, the trigger state contains the metadata of all the files processed till then (and not just the files modified). So for monitoring a folder with large number of files its expected that the host setting is updated to higher value so that the trigger size can accumulate the metdata of the files.

Regarding the use case mentioned, we aleady have an input parameter `maxFilecount` on the trigger to limit the number of files for a trigger run.

**Reply by raunaknarooka** [reply]
But if i ignore the maxfilecount parameter as it states in the connector, the behaviour is to pick up all files. But the default limit still exists for upto 1000 files because of the setting: ServiceProviders.MaximumAllowedTriggerStateSizeInKB.

I am not sure how many files can come in the service window, so it becomes  a bit difficult to set an exact value here.
